services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java

```diff
@@ -19,29 +23,45 @@
  */
 @Configuration
 public class TablesMvcConfigurer implements WebMvcConfigurer {
-
   @Autowired private ClusterProperties clusterProperties;
+  OpenAPI openAPI =
+      new OpenAPI()
+          .info(
+              new Info()
+                  .title("OpenHouse Tables APIs")
+                  .description("API description for OpenHouse Tables API")
+                  .termsOfService("http://swagger.io/terms")
+                  .version("v0.1")
+                  .license(new License().name("Apache 2.0").url("http://springdoc.org")));
 
   public void addInterceptors(InterceptorRegistry registry) {
-    if (clusterProperties.getClusterSecurityTokenInterceptorClassname() != null) {
-      Class<?> coordinatorClass =
-          ReflectionUtils.loadIfPresent(
-              clusterProperties.getClusterSecurityTokenInterceptorClassname(),
-              getClass().getClassLoader());
-      if (coordinatorClass != null) {
-        Optional<Constructor<?>> cons = ReflectionUtils.findConstructor(coordinatorClass);
-        if (cons.isPresent()) {
-          try {
-            registry
-                .addInterceptor((HandlerInterceptor) cons.get().newInstance())
-                .addPathPatterns("/**")
-                .excludePathPatterns("/actuator/**", "/**/api-docs/**", "/**/swagger-ui/**");
-          } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
-            throw new RuntimeException(
-                "Unable to install the configured Request Interception Handler", e);
-          }
-        }
-      }
-    }
+    Optional<HandlerInterceptor> handlerInterceptor =
+        ClusterPropertiesUtil.getClusterSecurityTokenInterceptor(clusterProperties);
+    handlerInterceptor.ifPresent(
+        interceptor -> {
+          registry
+              .addInterceptor(interceptor)
+              .addPathPatterns("/**")
+              .excludePathPatterns("/actuator/**", "/**/api-docs/**", "/**/swagger-ui/**");
+          addOpenAPISecurityScheme();
```

Comment:
if handlerinterceptor is precent, do both interceptor and add security to openapi